src/configuration/FirestoreConfigurationContext.tsx

```diff
@@ -0,0 +1,161 @@
+  defaultTariffZones,
+  defaultUserProfiles,
+} from '@atb/reference-data/defaults';
+
+type ConfigurationContextState = {
+  preassignedFareproducts: PreassignedFareProduct[];
+  tariffZones: TariffZone[];
+  userProfiles: UserProfile[];
+};
+
+const defaultConfigurationContextState: ConfigurationContextState = {
+  preassignedFareproducts: [],
+  tariffZones: [],
+  userProfiles: [],
+};
+
+const FirestoreConfigurationContext = createContext<ConfigurationContextState>(
+  defaultConfigurationContextState,
+);
+
+export const FirestoreConfigurationContextProvider: React.FC = ({children}) => {
+  const [preassignedFareproducts, setPreassignedFareproducts] = useState(
+    defaultPreassignedFareProducts,
+  );
+  const [tariffZones, setTariffZones] = useState(defaultTariffZones);
+  const [userProfiles, setUserProfiles] = useState(defaultUserProfiles);
+
+  useEffect(() => {
+    firestore()
+      .collection('configuration')
+      .onSnapshot(
+        (snapshot) => {
+          const preassignedFareproducts =
+            getPreassignedFarecontractsFromSnapshot(snapshot);
+          if (preassignedFareproducts) {
+            setPreassignedFareproducts(preassignedFareproducts);
+          }
+
+          const tariffZones = getTariffZonesFromSnapshot(snapshot);
+          if (tariffZones) {
+            setTariffZones(tariffZones);
+          }
+
+          const userProfiles = getUserProfilesFromSnapshot(snapshot);
+          if (userProfiles) {
+            setUserProfiles(userProfiles);
+          }
+        },
+        (error) => {
+          Bugsnag.leaveBreadcrumb(
+            `Firebase Error when fetching Configuration from Firestore`,
+            error,
+          );
+        },
+      );
+  }, []);
+
+  const memoizedState = useMemo(() => {
+    return {
+      preassignedFareproducts,
+      tariffZones,
+      userProfiles,
+    };
+  }, [preassignedFareproducts, tariffZones, userProfiles]);
+
+  return (
+    <FirestoreConfigurationContext.Provider value={memoizedState}>
+      {children}
+    </FirestoreConfigurationContext.Provider>
+  );
+};
+
+export function useFirestoreConfiguration() {
+  const context = useContext(FirestoreConfigurationContext);
+  if (context === undefined) {
+    throw new Error(
+      'useConfigurationContext must be used within an useConfigurationContextProvider',
+    );
+  }
+  return context;
+}
+
+function getPreassignedFarecontractsFromSnapshot(
+  snapshot: FirebaseFirestoreTypes.QuerySnapshot,
+): PreassignedFareProduct[] | null {
+  const preassignedFareproductsFromFirestore = snapshot.docs
+    .find((doc) => doc.id == 'referenceData')
+    ?.get<string>('preassignedFareProducts_v2');
+
+  try {
+    if (preassignedFareproductsFromFirestore) {
+      return JSON.parse(
+        preassignedFareproductsFromFirestore,
+      ) as PreassignedFareProduct[];
+    }
+  } catch (error) {
+    Bugsnag.leaveBreadcrumb(
+      'Error parsing preassignedFareproducts from FireStore',
```

Comment:
The message error should change to `Error parsing preassignedFareproducts in v2 from FireStore`? Not sure about this version 2 specification in the error message!

Reply:
I don't think we actually need to version the data.
I'm not sure why data is named v2 in firestore, probably just for comparison/reference

Reply:
This is now changed to `Bugsnag.notify` and not as relevant. 

But regarding versioning: The versioning is actually important. We made some improvements to the reference data structure for `preassignedFareproducts` and made a v2, but old versions of the app are still fetching the old version (v1) of the reference data.